### PR TITLE
tiny accuracy improvements for spu group joins

### PR DESCRIPTION
**Pull request by elad335**

### CODE REVIEW

**Comment by darkdragonz12** [comment]
this breaks for my naruto ultimate ninja storm 2, blazblue central fiction.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2577926/RPCS3.log.gz)

**Comment by darkdragonz12** [comment]
has returned to normal with the last pr, great job! :D

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 1]
```diff
@@ -488,7 +488,10 @@ void spu_thread::cpu_stop()
 		if (verify(HERE, group->running--) == 1)
 		{
 			// Notify on last thread stopped
-			group->mutex.lock_unlock();
+			{
```
Comment (AniLeo) [inline rpcs3/Emu/Cell/SPUThread.cpp]: Why the extra brackets?

File: rpcs3/Emu/Cell/lv2/sys_spu.cpp [hunk 2]
```diff
@@ -618,14 +606,11 @@ error_code sys_spu_thread_group_terminate(u32 id, s32 value)
...
 
-	group->run_state = SPU_THREAD_GROUP_STATUS_INITIALIZED;
 	group->exit_status = value;
 	group->join_state |= SPU_TGJSF_TERMINATED;
 
 	// Wait until the threads are actually stopped
-	const u64 last_stop = group->stop_count - !group->running;
-
-	while (group->stop_count == last_stop)
+	while (group->run_state != SPU_THREAD_GROUP_STATUS_INITIALIZED)
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_spu.cpp]: Don't remove stop_count.

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 3]
```diff
@@ -487,8 +487,8 @@ void spu_thread::cpu_stop()
 		if (verify(HERE, group->running--) == 1)
 		{
 			// Notify on last thread stopped
-			group->stop_count++;
-			group->mutex.lock_unlock();
+			std::lock_guard lock(group->mutex);
+			group->run_state = SPU_THREAD_GROUP_STATUS_INITIALIZED;
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/SPUThread.cpp]: Mutex should be unlocked before notification.

**Pull request merged**